app/src/main/kotlin/com/github/oryanmat/trellowidget/data/remote/ApiResponse.kt

```diff
@@ -0,0 +1,9 @@
+package com.github.oryanmat.trellowidget.data.remote
+
+
+sealed class ApiResponse<out T> {
+
+    data class Success<out T>(val data: T) : ApiResponse<T>()
+
+    data class Error<out T>(val error: String) : ApiResponse<T>()
```

Comment:
Because of the type erasure in Java, we can't conveniently use a sealed interface with generic functions. Instead, I used a generic sealed class for this.

Reply:
The limitation regarding generics is true for an abstract class the same way it is true for an interface. 
So why not use an interface?